content/docs/typechecking-with-proptypes.md

```diff
@@ -169,3 +169,59 @@ ReactDOM.render(
 ```
 
 The `defaultProps` will be used to ensure that `this.props.name` will have a value if it was not specified by the parent component. The `propTypes` typechecking happens after `defaultProps` are resolved, so typechecking will also apply to the `defaultProps`.
+
+### Nested React Elements (e.g. children)
+
+If you want to validate nested elements, such as the children of your component, you may need to use a nested shape on `props` if your nested children are React elements:
+
+```javascript
+import PropTypes from 'prop-types';
+import SubComponent from './my-custom-components/SubComponent'
+
+class MyComponent extends React.Component {
+  render() {
+    // This must be exactly one element or it will warn.
+    const child = this.props.children;
+    return (
+      // Condition 1
+      <div>
+        {child}
+      </div>
+      // Condition 2
+      <SubComponent 
+        containerElement={child.props.containerElement}
+        primaryText={child.props.label}
+      />
+    );
+  }
+}
```

Comment:
This isn't valid JSX ^ Maybe you could change it to something like:

```jsx
class MyComponent extends React.Component {
  render() {
    // This must be exactly one element or it will warn.
    const child = this.props.children;

    if (someConditional) {
      return (
        <div>
          {child}
        </div>
    	);
    } else {
      return (
        <SubComponent 
          containerElement={child.props.containerElement}
          primaryText={child.props.label}
        />
      );
    }
  }
}
```